### ISSUE TITLE
Offscreen rendering feature 

### ISSUE THREAD

**Issue by PooyaEimandar** [tauri-apps/wry#391]
Hello
Thank you for contributing to this amazing project. Does wry support render to texture? I mean something like offscreen rendering?

**Comment by wusyong** [comment]
It's possible to do it but you would need to do it yourself with tao window creation.
But I suppose it will render offscreen by default.

**Comment by wusyong** [comment]
After searching a bit, I think this feature is possible on Linux, and [macOS ](https://stackoverflow.com/questions/30149373/render-off-screen-wkwebview-into-nsimage)which both use webkit underneath.
Windows, however, is still an [open issue](https://github.com/MicrosoftEdge/WebView2Feedback/issues/547).

**Comment by anlumo** [comment]
It's possible on Windows 10 and 11: https://github.com/jnschulze/flutter-webview-windows

**Comment by TrueDoctor** [comment]
Could this workaround be integrated? @wusyong 
https://github.com/MicrosoftEdge/WebView2Feedback/issues/547#issuecomment-1925031256

**Comment by TapGhoul** [comment]
I've been vaguely following the deal with offscreen rendering with WebView2 for embedding within bevy - I'm curious if the technique described in https://gist.github.com/pabloko/5b5bfb71ac52d20dfad714c666a0c428 would work to close the gap here.

Tldr (from their readme, but terser) it uses DirectComposition by calling `CreateCoreWebView2CompositionController()` instead of `CreateCoreWebView2Controller()`, and creates a `Direct3D11CaptureFramePool` to get the rendered frames back out via `GraphicsCaptureItem::CreateFromVisual()`.

It's more efficient than using a hidden window and capturing it (as far as I can tell, not sure how true this is, but windows graphics capture has given me pain in past), and is natively supported by WV2 - it's just not as well documented, but it's not an API that's marked as experimental.
It seems similar to how the WinUI3 stuff works as mentioned in the issue @TrueDoctor linked, but in this case just giving us direct access to the composition result rather than having it go through the UI framework first.

I haven't tested performance here - windows is not my primary build target at the moment - but it might be worth poking at as a way to finally get this issue resolved.

This would also mean a lack of support for things like drag & drop and such, but then that's an issue with offscreen rendering as a whole (given we are inherently lacking a window to capture the drop event on), so I can't imagine it'd be a blocker in of itself.
